src/components/admin/lib/article-validators.js

```diff
@@ -0,0 +1,24 @@
+export function checkForHttp(html) {
+  if (/http:\/\/.*\..{3,}/.test(html) || /http(?!s)/.test(html)) {
+    return false;
+  } else if (/https:\/\/.*\..{3,}/.test(html)) {
+    return true;
+  }
+  return false;
+}
+// original if statement for first function (/http(?!s)/.test(html))
+// http(?!s):\/\/\..*\..{3,}
+// has a non https link in the body
+// we want the function to equal true
+// if the function does have an s, the function returns true
+
+export function checkForAbsoluteUrlRegex(html) {
```

Comment:
We wanted this function to return the actual url right?

Reply:
Nop, If I remember correctly, it simply returns a boolean. The actual URL is returned by another function in the MainIssueController

Reply:
Yeah, but it makes us duplicate the regex twice when we could just have it in one place. When we have it in two places it opens us up to errors of just changing it one place and forgetting it another place and introducing weird bugs. Also this means that the stuff in the IssueController isn't tested. In the IssueController you could just check whether it returns a url or not, and if it does that means it wasn't absolute